### PR TITLE
Emulate superagent's automatic encoding of JSON body

### PR DESCRIPTION
**Pull request by wyuenho**
This is rather urgent as some internal code just assumes the body will be automatically encoded

### CODE REVIEW

**Comment by wyuenho** [comment]
@jculvey @tajo @manatarms @jwineman I just simplified the code a lot, can you re-review again 🙏🙏🙏🙏🙏🙏🙏🙏

File: packages/cf-util-http/src/http.js [hunk 1]
```diff
@@ -123,14 +127,21 @@ export function request(method, url, opts, callback) {
   url = opts.url;
   callback = opts.callback;
 
+  // Emulate superagent's ability to automatically encode JSON body and set the
+  // Content-Type
+  if (BODYLESS_METHODS.every(m => method.toUpperCase() !== m) && opts.body) {
+    opts.body = JSON.stringify(opts.body);
+    opts.headers = new Headers(opts.headers || {});
+    opts.headers.set('Content-Type', 'application/json');
```
Comment (jwineman) [inline packages/cf-util-http/src/http.js]: Does this replace the `Content-Type` header if it already exists or add a second one?  I can't find docs for `.set()` they all link to `Set()`(Data structure)

Comment (wyuenho) [inline packages/cf-util-http/src/http.js]: https://developer.mozilla.org/en-US/docs/Web/API/Headers/set

Comment (jwineman) [inline packages/cf-util-http/src/http.js]: Doh.  Thanks @wyuenho 
https://developer.mozilla.org/en-US/docs/Web/API/Headers/set

Comment (wyuenho) [inline packages/cf-util-http/src/http.js]: I'll make it conditional

File: packages/cf-util-http/src/http.js [hunk 2]
```diff
@@ -112,6 +114,8 @@ export function request(method, url, opts, callback) {
     opts = {};
   }
 
+  opts = Object.assign({}, opts);
```
Comment (sejoker) [inline packages/cf-util-http/src/http.js]: love this change.

**Pull request merged**